core/src/main/scala/org/broadinstitute/dsde/rawls/workspace/WorkspaceService.scala

```diff
@@ -2193,6 +2195,78 @@ class WorkspaceService(
       throw new RawlsExceptionWithErrorReport(errorReport = err)
     }
   }
+
+  def addAuthDomainGroups(workspaceName: WorkspaceName,
+                          newAuthDomainGroups: Set[String],
+                          parentContext: RawlsRequestContext = ctx
+  ): Future[Unit] = for {
+    // relies on sam calls to check permissions
+    // if no auth domain update is required, caller only requires read_auth_domain permission
+    // if an auth domain update is required, caller requires update_auth_domain permission
+    workspace <- getV2WorkspaceContext(workspaceName)
+    existingAuthDomain <- samDAO
+      .getResourceAuthDomain(SamResourceTypeNames.workspace, workspace.workspaceId, parentContext)
+
+    _ <-
+      if ((newAuthDomainGroups -- existingAuthDomain).nonEmpty) {
```

Comment:
since we just talked about it … is this case-sensitive and could cause weird behavior?